### ISSUE TITLE
Halt on `lib/src/db/mod.rs/Pool/close()/self.reads.close().await`

### ISSUE THREAD

**Issue by inetic** [equalitie/ouisync#107]
Reproducible from the `failed_block_other_peer` test in `lib/src/network/tests.rs` but only after many iterations and while running `stress-ng` (the exact parameters I used are `stress-ng -f 30 -i 1000 -m 100` but this can vary).

I think the test in question sometimes halts at other places as well.

**Comment by madadam** [comment]
`Pool::close` waits until all db connections are released back to the pool, so a hang there suggest a connection is being stuck somewhere. Possibly a deadlock?

**Comment by inetic** [comment]
So it seems that the halt comes from [here](https://github.com/equalitie/ouisync/blob/e5f73d78a9c501e1dcb593d203c44becc037172d/lib/src/network/tests.rs#L340) and I suspect that it's because [`conn`](https://github.com/equalitie/ouisync/blob/e5f73d78a9c501e1dcb593d203c44becc037172d/lib/src/network/tests.rs#L331) is still alive. If that's the issue then adding `drop(conn)` on line 335 should do the trick, testing it now...

**Comment by inetic** [comment]
I think this is closed now by https://github.com/equalitie/ouisync/commit/9cc96fd254f900906ad8877a9c3a5e0decf7e03f. The tests still occasionally fails down the line, but that's another issue.
